### PR TITLE
task/PVA-930 Create a new endpoint for moving assets within a site

### PR DESCRIPTION
**Pull request by jinglitw**
# API Pull Request
## VERSION Number
- 2.3.0
## Ticket url
- https://chillibean.atlassian.net/browse/PVA-930
## Changes Made
- create a new endpoint for moving assets within a site
## Checklist
Please verify this checklist so that you are sure this is ready for review.

- [ ]  The only file that has been changed is schema.json - unless you needed to add some new functionality (if so, please update generator repo too)
- [ ]  Updated package.json & schema.json versions - they need to be the same
- [ ]  All objects have a properites list, required array and additionalProperties set to false

### CODE REVIEW

**Comment by robertozmc** [comment]
Previous PR with comments for history reasons
https://github.com/Chillibean/subjects-api-specifications/pull/43

File: public/refs/components/schemas/AssetsMove.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+  "type": "object",
+  "properties": {
+    "assets": {
```
Comment (robertozmc) [inline public/refs/components/schemas/AssetsMove.json]: https://github.com/Chillibean/subjects-api-specifications/pull/43#discussion_r1824030821

File: public/refs/components/schemas/AssetsMove.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      "example": "E0010023"
+    },
+    "study_event_identifier": {
+      "type": "string",
+      "description": "The study event identifier",
+      "example": "VISIT1"
+    },
+    "study_procedure_identifier": {
+      "type": "string",
+      "description": "The asset study procedure identifier",
```
Comment (robertozmc) [inline public/refs/components/schemas/AssetsMove.json]: https://github.com/Chillibean/subjects-api-specifications/pull/43#discussion_r1824032472

File: public/refs/paths/accounts_{account_id}_sites_{site_identifier}_subjects_{subject_number}_study_events_{study_event_identifier}_study_procedures_{study_procedure_identifier}_assets_move.json [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        "application/json": {
+          "schema": {
+            "type": "object",
+            "properties": {
+              "authenticity_token": {
+                "type": "string",
+                "description": "The authenticity token",
+                "example": "valid_token"
+              },
+              "items": {
```
Comment (robertozmc) [inline public/refs/paths/accounts_{account_id}_sites_{site_identifier}_subjects_{subject_number}_study_events_{study_event_identifier}_study_procedures_{study_procedure_identifier}_assets_move.json]: its not true, the `AssetsMove` content should be part of the `properties` and not be nested into `items`

**Review by robertozmc — CHANGES_REQUESTED** [review]

File: public/refs/components/schemas/AssetsMove.json (hunk 1 above)
Comment (jinglitw) [reply public/refs/components/schemas/AssetsMove.json]: ok I missunderstood

File: public/refs/paths/accounts_{account_id}_sites_{site_identifier}_subjects_{subject_number}_study_events_{study_event_identifier}_study_procedures_{study_procedure_identifier}_assets_move.json (hunk 3 above)
Comment (robertozmc) [reply public/refs/paths/accounts_{account_id}_sites_{site_identifier}_subjects_{subject_number}_study_events_{study_event_identifier}_study_procedures_{study_procedure_identifier}_assets_move.json]: it means that all the properties from `AssetMove` ref should be put next to `authenticity_token` and **not** inside `items` property

What you have now:

```json
{
  "body": {
    "authenticity_token": "valid_token",
    "items": {
      "assets": [1, 4, 100],
      "subject_number": "E0010023",
      "study_event_identifier": "VISIT1",
      "study_procedure_identifier": "NSTAR",
      "reason_for_moving": "Mistake"
    }
  }
}
```

vs what you should have:

```json
{
  "body": {
    "authenticity_token": "valid_token",
    "assets": [1, 4, 100],
    "subject_number": "E0010023",
    "study_event_identifier": "VISIT1",
    "study_procedure_identifier": "NSTAR",
    "reason_for_moving": "Mistake"
  }
}
```

File: public/refs/paths/accounts_{account_id}_sites_{site_identifier}_subjects_{subject_number}_study_events_{study_event_identifier}_study_procedures_{study_procedure_identifier}_assets_move.json [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+                "$ref": "../components/schemas/AssetsMove.json"
```
Comment (robertozmc) [inline public/refs/paths/accounts_{account_id}_sites_{site_identifier}_subjects_{subject_number}_study_events_{study_event_identifier}_study_procedures_{study_procedure_identifier}_assets_move.json]: also do we even need that AssetsMove schema? we haven't used that for requests, only responses

File: public/refs/paths/accounts_{account_id}_sites_{site_identifier}_subjects_{subject_number}_study_events_{study_event_identifier}_study_procedures_{study_procedure_identifier}_assets_move.json [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+        "schema": {
+          "type": "string",
+          "example": "PROC1"
+        }
+      }
+    ],
+    "requestBody": {
+      "content": {
+        "application/json": {
+          "schema": {
```
Comment (robertozmc) [inline public/refs/paths/accounts_{account_id}_sites_{site_identifier}_subjects_{subject_number}_study_events_{study_event_identifier}_study_procedures_{study_procedure_identifier}_assets_move.json:68]: missing `required` property for all fields in the body and `additionalProperties: false`

Comment (jinglitw) [reply public/refs/paths/accounts_{account_id}_sites_{site_identifier}_subjects_{subject_number}_study_events_{study_event_identifier}_study_procedures_{study_procedure_identifier}_assets_move.json:68]: right, we have that for updating endpoints

File: public/refs/paths/accounts_{account_id}_sites_{site_identifier}_subjects_{subject_number}_study_events_{study_event_identifier}_study_procedures_{study_procedure_identifier}_assets_move.json (hunk 4 above)
Comment (jinglitw) [reply public/refs/paths/accounts_{account_id}_sites_{site_identifier}_subjects_{subject_number}_study_events_{study_event_identifier}_study_procedures_{study_procedure_identifier}_assets_move.json]: ok I misunderstood before, will remove this

File: public/refs/paths/accounts_{account_id}_sites_{site_identifier}_subjects_{subject_number}_study_events_{study_event_identifier}_study_procedures_{study_procedure_identifier}_assets_move.json (hunk 5 above)
Comment (jinglitw) [reply public/refs/paths/accounts_{account_id}_sites_{site_identifier}_subjects_{subject_number}_study_events_{study_event_identifier}_study_procedures_{study_procedure_identifier}_assets_move.json:68]: I found we sometimes added sometimes didn't. We need to unify them in the future.

File: public/refs/paths/accounts_{account_id}_sites_{site_identifier}_subjects_{subject_number}_study_events_{study_event_identifier}_study_procedures_{study_procedure_identifier}_assets_move.json [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+                "example": "NSTAR"
+              },
+              "reason_for_moving": {
+                "type": "string",
+                "description": "The reason for moving the assets",
+                "example": "Mistake"
+              }
+            },
+            "required": [
+              "authenticity_token"
```
Comment (robertozmc) [inline public/refs/paths/accounts_{account_id}_sites_{site_identifier}_subjects_{subject_number}_study_events_{study_event_identifier}_study_procedures_{study_procedure_identifier}_assets_move.json]: what about adding other required fields now?

File: public/refs/paths/accounts_{account_id}_sites_{site_identifier}_subjects_{subject_number}_study_events_{study_event_identifier}_study_procedures_{study_procedure_identifier}_assets_move.json (hunk 5 above)
Comment (robertozmc) [reply public/refs/paths/accounts_{account_id}_sites_{site_identifier}_subjects_{subject_number}_study_events_{study_event_identifier}_study_procedures_{study_procedure_identifier}_assets_move.json:68]: can update it for this one now though

**Review by robertozmc — CHANGES_REQUESTED** [review]
one more comment

**Review by robertozmc — APPROVED** [review]

**Pull request merged**